src/functional/iterator/embedded.py

```diff
@@ -143,8 +142,33 @@ def __array__(self) -> np.ndarray:
         ...
 
 
-def _is_column(v: Any) -> TypeGuard[Column]:
-    return isinstance(v, np.ndarray)
+class Column(numpy.lib.mixins.NDArrayOperatorsMixin):
+    def __init__(self, kstart: int, data: np.ndarray):
+        self.kstart = kstart
+        self.data = data
+
+    def __getitem__(self, i: int):
+        return self.data[i - self.kstart]
+
+    def tuple_get(self, i: int):
+        if self.data.dtype.names is not None:
+            return Column(self.kstart, self.data[f"f{i}"])
+        else:
+            return Column(self.kstart, self.data[i, ...])
+
+    def __setitem__(self, i: int, v: Any):
+        self.data[i - self.kstart] = v
+
+    def __array__(self, dtype=None) -> np.ndarray:
+        return self.data if dtype is None or dtype == self.data.dtype else self.data.astype(dtype)
+
+    def __array_function__(self, func, types, args, kwargs):
+        assert all(issubclass(t, self.__class__) for t in types)
+        return self.__class__(self.kstart, func(*(arg.data for arg in args), **kwargs))
+
+
+def _make_column(column_range: range, dtype=np.dtype):
+    return Column(min(column_range), np.zeros(len(column_range), dtype=dtype))
```

Comment:
```suggestion
    return Column(column_range.start, np.zeros(len(column_range), dtype=dtype))
```
Have you checked that min does not iterate though the entire column? In that case this is just a suggestion. You could also use `column_range[0]`